surprise/builtin_datasets.py

```diff
@@ -19,8 +21,12 @@ def get_dataset_dir():
 
     folder = os.environ.get('SURPRISE_DATA_FOLDER', os.path.expanduser('~') +
                             '/.surprise_data/')
-    if not os.path.exists(folder):
+    try:
         os.makedirs(folder)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            # reraise exception if folder exists and creation failed.
```

Comment:
don't we ignore the exception precisely when the folder exists?

Reply:
@lrebscher ?

Reply:
no need to push @MeTaNoV. Thanks.

Reply:
@NicolasHug, I always push when it's a red BLOCKER! ;)

Reply:
This is open source work. We're not paid for this.

Please reconsider your approach.

Reply:
all GOOD! :)

Reply:
@NicolasHug , happy national day? :dancers: ;)

Reply:
and for the story, I am taking over the legacy of @lrebscher

Reply:
@NicolasHug indeed! 

It should have been:
```suggestion
            # reraise exception if folder does not exist and creation failed.
```